### ISSUE TITLE
Document how to debug performance of prow job

### ISSUE THREAD

**Issue by serathius** [etcd-io/etcd#18707]
### What would you like to be added?

In robustness tests we noticed a series of flakes and we wanted to eliminate performance as a factor. @ah8ad3 has reached out to Kubernetes prow and found interesting dashboards:
https://monitoring-gke.prow.k8s.io/d/96Q8oOOZk/builds?orgId=1&refresh=30s&var-org=etcd-io&var-repo=etcd&var-job=ci-etcd-robustness-main-amd64&var-build=All&from=now-7d&to=now

We should preserve this knowledge and document for future how etcd contributors can navigate performance of prow jobs. Possibly a new section in etcd contributor documentation about prow that either links to the dashboard or some prow documentation.

### Why is this needed?

Make it easier for others to debug etcd job performance issues.

**Comment by serathius** [comment]
cc @ah8ad3

**Comment by ivanvc** [comment]
I agree. There's also an EKS dashboard, but they are not documented anywhere in the infra testing repo, I also found about it by asking on Slack

**Comment by CFly17** [comment]
I am interested in learning more about and contributing to etcd!

Are you thinking of a new doc (like `etcd/Documentation/contributor-guide/prow_jobs.md`) with sections like:

- Introduction to Prow
- How Prow is Used for etcd Testing
- Creating/Navigating Performance Dashboard (Grafana)
- Interpreting Metrics (link to [prow documentation](https://github.com/kubernetes-sigs/prow/blob/main/site/content/en/docs/metrics/_index.md))
- Troubleshooting Common Issues (e.g.?)
- etc.

Please let me know if I can help! :)

**Comment by ivanvc** [comment]
Hi, @CFly17, thanks for volunteering!

> Are you thinking of a new doc (like etcd/Documentation/contributor-guide/prow_jobs.md) with sections like:

Yes, we need a new document, and your suggestion sounds good. I think we can start with:

* Introduction to Prow 
* How Prow is Used for etcd Testing - And where the jobs are located, i.e., test-infra repository, in the `config/jobs/etcd directory`
* Navigating Performance Dashboard (Grafana) (just navigating. I don't think we have an intention to create dashboards at this moment
* Selecting our jobs, explaining the different kinds of workloads/jobs, i.e., robustness, integration, static check, etc. (at a high level).
* Interpreting Metrics

Feel free to open a pull request (or draft pull request). Would you like to be assigned? If so, just send a comment with `/assign` (in plain text, w/o code backticks), and the k8s bot will assign it to you.

**Comment by ivanvc** [comment]
We reviewed this task in the triage meeting. @CFly17, are you still working on this one, or can we reassign it? Thanks.

**Comment by CFly17** [comment]
Please forgive me @ivanvc  -- I tried reading up on the Raft algorithm and got lost in the sauce. I hope to contribute to **etcd** sometime, but I fear I may be in over my head with this one.

**Comment by ivanvc** [comment]
@joshjms, should we unassign this issue from you?

**Comment by joshjms** [comment]
Oops, yes sure, thanks!

**Comment by ronaldngounou** [comment]
Hi, This is a good opportunity to improve my understanding of Prow and test suites.
I would love to pick this up. 

In the meantime, do you have any existing resources you would recommend for:
- Navigating Performance Dashboard (Grafana) (just navigating. I don't think we have an intention to create dashboards at this moment
- Explaining the different kinds of workloads/jobs, i.e., robustness, integration, static check, etc. (at a high level).
- Interpreting Metrics

Thank you in advance.

**Comment by ronaldngounou** [comment]
Hi @ivanvc, 
I made some progress and I had a few questions:

1. Navigating Performance Dashboard (Grafana):
Is there any reasons why GKE dashboard have some data populated but not EKS dashboard? Do you know how these dashboard get populated? For this documentation, should I focus on GKE dashboard then?

2. Are there any specific metrics  besides CPU, Memory that would you like to be analysed?

**Comment by pratik-mahalle** [comment]
Hey @ivanvc, I would love to take this up. Can you please explain me this if possible?
Thanks

**Comment by ronaldngounou** [comment]
@pratik-mahalle I'm already working on this task and I have a draft PR ready [#20796](https://github.com/etcd-io/etcd/pull/20796)

**Comment by ivanvc** [comment]
Hey, @ronaldngounou, thanks for picking up this issue.

>     * Navigating Performance Dashboard (Grafana) (just navigating. I don't think we have an intention to create dashboards at this moment

You can find the Grafana dashboards in the comment https://github.com/etcd-io/etcd/issues/18707#issuecomment-2402754334. We don't have any additional resources.
 
>     * Explaining the different kinds of workloads/jobs, i.e., robustness, integration, static check, etc. (at a high level).

You can take a look at any PR, and you will see the jobs that we're currently running. We have unit, integration, e2e, and robustness tests. We also have static check analysis, and other miscellaneous tests like release script tests, govulncheck, etc.

>     * Interpreting Metrics

Please refer to https://github.com/etcd-io/etcd/issues/18707#issuecomment-2434164966 / https://github.com/kubernetes-sigs/prow/blob/main/site/content/en/docs/metrics/_index.md

@pratik-mahalle, sorry, @ronaldngounou already self-assigned himself this issue :)

**Comment by ronaldngounou** [comment]
Merged in PR https://github.com/etcd-io/etcd/pull/20796
